build.make

```diff
@@ -69,12 +69,15 @@ endif
 # toolchain.
 BUILD_PLATFORMS =
 
+# Add go ldflags using LDFLAGS at the time of compilation.
+LDFLAGS ?= -X main.version=$(REV) -extldflags "-static" 
```

Comment:
When using `make LDFLAGS=<something>`, the user has to remember to include these defaults in `<something>`. This looks a bit error-prone.

What about this?

```
# LDFLAGS can be set with "make LDFLAGS=..." to add custom link flags.
LDFLAGS =

# Some link flags are used automatically. This can be suppressed with "make FULL_LDFLAGS=..."
FULL_LDFLAGS =  -X main.version=$(REV) -extldflags "-static" $(LDFLAGS)

... -ldflags '$(FULL_LDFLAGS)' ...
```

Reply:
@pohly, thanks for reviewing this. Will this suggested change allow adding extldflags ? The idea behind this PR is to make sure, we can add any ldflags we require.

Reply:
I have added a few more variables to make sure everything is configurable.